packages/musap-rn-kms/src/agent/MusapKeyManagerSystem.ts

```diff
@@ -0,0 +1,127 @@
+import { IKey, ManagedKeyInfo, MinimalImportableKey, TKeyType } from '@veramo/core';
+import {
+  KeyAlgorithmType,
+  KeyGenReq,
+  MusapKey,
+  MusapModuleType, signatureAlgorithmFromKeyAlgorithm,
+  SignatureAlgorithmType,
+  SignatureFormat,
+  SignatureReq,
+} from '@sphereon/musap-react-native'
+import { SscdType } from '@sphereon/musap-react-native/src/types/musap-types';
+import { KeyManagementSystem } from '@veramo/kms-local';
+import { AbstractPrivateKeyStore } from '@veramo/key-manager';
+import 'react-native-get-random-values'
+import { v4 as uuid } from 'uuid';
+import { TextDecoder } from 'text-encoding';
+import Debug from 'debug'
+
+const debug = Debug('sphereon:musap-rn-kms')
+
+export class MusapKeyManagementSystem extends KeyManagementSystem {
+  private musapKeyStore: MusapModuleType;
+  private sscdType: SscdType;
+
+  constructor(keyStore: MusapModuleType, sscdType?: SscdType) {
+    super(keyStore as unknown as AbstractPrivateKeyStore);
+    this.musapKeyStore = keyStore;
+    this.sscdType = sscdType ? sscdType : 'TEE';
+    this.musapKeyStore.enableSscd(this.sscdType)
```

Comment:
How does this work when enabling a SSCD that might not yet be registered?

Reply:
according to @sanderPostma this is the registering of the sscd